tests/integration/rosetta/rosetta_test.go

```diff
@@ -0,0 +1,256 @@
+
+	svc, db, err := setupDB()
+	if err != nil {
+		log.Fatalf("could not perform setup for test: %v", err)
+	}
+
+	rosettaSvc = svc
+
+	defer db.Close()
+
+	m.Run()
+}
+
+func TestGetBalance(t *testing.T) {
+
+	tests := []struct {
+		name string
+
+		request rosetta.BalanceRequest
+
+		wantStatusCode int
+		wantBalance    string
+		wantHandlerErr assert.ErrorAssertionFunc
+	}{
+		// TODO: use a different 'valid balance request' - one that has an actual, non-zero balance
+		{
+			name:           "valid balance request",
+			request:        getBalanceRequest("8c5303eaa26202d6", 0, "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"),
+			wantBalance:    "0",
+			wantStatusCode: http.StatusOK,
+			wantHandlerErr: assert.NoError,
+		},
+		{
+			name:           "empty balance request",
+			request:        rosetta.BalanceRequest{},
+			wantStatusCode: http.StatusUnprocessableEntity,
+			wantHandlerErr: assert.Error,
+		},
+		{
+			name:           "block hash and height mismatch",
+			request:        getBalanceRequest("8c5303eaa26202d6", 99, "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"),
+			wantStatusCode: http.StatusUnprocessableEntity,
+			wantHandlerErr: assert.Error,
+		},
+		{
+			name:           "invalid account address",
+			request:        getBalanceRequest("invalid_address", 0, "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"),
+			wantStatusCode: http.StatusUnprocessableEntity,
+			wantHandlerErr: assert.Error,
+		},
+	}
+
+	for _, test := range tests {
+
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			// prepare request payload (JSON)
+			enc, err := json.Marshal(test.request)
+			require.NoError(t, err)
+
+			// create request
+			req := httptest.NewRequest(http.MethodPost, "/account/balance", bytes.NewReader(enc))
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			rec := httptest.NewRecorder()
+
+			ctx := echo.New().NewContext(req, rec)
+
+			// execute the request
+			err = rosettaSvc.Balance(ctx)
+			test.wantHandlerErr(t, err)
+
+			// validate response data
+
+			// validate status code
+			if test.wantStatusCode != http.StatusOK {
+
+				// ugly workaround for HTTP status code set by router vs set on context
+				e, ok := err.(*echo.HTTPError)
+				require.True(t, ok)
+				assert.Equal(t, test.wantStatusCode, e.Code)
+
+				// nothing more to do, response validation should only be done for '200 OK' responses
+				return
+			}
+
+			assert.Equal(t, test.wantStatusCode, rec.Result().StatusCode)
+
+			// unpack response
+			var balanceResponse rosetta.BalanceResponse
+			require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &balanceResponse))
+
+			// verify that the block data matches the input data
+			assert.Equal(t, test.request.BlockID.Index, balanceResponse.BlockID.Index)
+			assert.Equal(t, test.request.BlockID.Hash, balanceResponse.BlockID.Hash)
+
+			// verify that we have at least one balance in the response
```

Comment:
Since this branched was spawned off of a fix branch that is now obsolete (`fix-cadence-script-execution`), I've moved this code to another branch, based on master, which is clean.

I've closed this PR and opened another one, with the same code (minor initialization code changes).

I have also addressed this comment in the code in the other PR.

Sorry for the mess now :)